VAMobile/documentation/docs/Engineering/FrontEnd/DevelopmentBestPractices.md

```diff
@@ -0,0 +1,135 @@
+# Development Best Practices
+
+Our team follows a set of best practices around tickets, branches, and PRs to ensure a smooth development process and high quality code. Here we'll describe the key points to keep in mind.
+
+## Tickets
+
+We use tickets to track all our engineering work. They precisely specify the work to be performed, allow stakeholders to track progress, and guide QA in assessing the resulting work.
+
+### Creating tickets
+
+Every ticket must contain at least the following information:
+
+- A detailed description of the work
+- Clearly specified Acceptance Criteria
+- Screenshots or links to mockups where applicable
+- All relevant labels ("front-end" for FE work, "Health" for Health and Benefits work, "Global" for global work, etc)
+- An epic relating to the ticket's larger context
+
+If possible, choose the relevant template for your ticket. For example,  if you're creating a bug ticket, choose the Bug Report Template and include severity in the title, labels, and body.
+
+### Working on tickets
+
+All engineering work requires a ticket.
+
+Every ticket must satisfy the requirements in the "Creating tickets" section above. If information is missing, add it or contact the relevant stakeholders.
+
+Before an engineer can begin work on a ticket, it must contain an estimate. Bugs get an initial estimate of 1 to cover a short investigation. Other tickets are left to the engineers' discretion.
+
+Once everything is in place, complete the following tasks:
+
+- Assign yourself
+- Assign the appropriate QA engineer if not already assigned
+- Move the ticket to In Progress
+- Create a branch as described below.
+
+Then you can begin work. Keep stakeholders up to date as you work by commenting on the ticket with any important details or questions. This will also preserve information so everyone can reference it in the future.
+
+On completion of your work, double check the following:
+
+- All items in Acceptance Criteria are implemented
+- Accessibility features are in place (test with iOS and Android screen readers)
+- Unit and E2E tests are updated or extended to cover your changes
+- Relevant documentation is updated to reflect the changes in your PR
+
+Then create a Pull Request. See the Pull Requests section below for more information.
+
+## Branches
+
+Every code change to the app takes place in a feature branch. We use feature-xl branches for extra-large changes.
```

Comment:
This would just be for feature work. I think we can kill the concept of `feature-xl` since it's just something we made up. Branch naming conventions should be like:

- `chore/1234-name-of-chore-being-done`
- `bugfix/1234-name-of-bug-being-done`
- `hotfix/1234-name-of-bug-being-done`
- `feature/1234-name-of-feature-being-done`

Some references about this:

- [Naming git branches](https://gist.github.com/jefffederman/1d492f98b8e3913a75ca)
- [Branch naming](https://dev.to/varbsan/a-simplified-convention-for-naming-branches-and-commits-in-git-il4)

Reply:
Makes sense to me, updated